### DISCUSSION TITLE
Feature request: Link to defined custom commands, or local shell command

### DISCUSSION THREAD

**Discussion by Fooseigh** [zim-desktop-wiki/zim-desktop-wiki#2118] [Q&A] [answered]
Hi,

I am looking for ways to call a local shell command from within the ZIM content window.

Situation: In my ZIM notebook, I keep phone numbers for specific contacts regarding the context. A custom command calls a shell script which passes a highlighted phone number to my ADSL/Phone router, which in turn initiates a VoiP call and passes it to my desktop. So I can call phone numbers from within ZIM.

However, I have to highlight the entire phone number first, then click on the custom command icon in the toolbar.

A link behind the \tel character would be a much more convenient way to do this, if it can call a shell command and pass a parameter, something like a shell:// URL for example.

Are there other ways to call external commands that I have missed?

ZIM is v0.65

Thanks & regards,
Hans

**Comment by introt** [comment]
Hi,

This topic has been previously discussed over at #513 - there's currently no support for URI:s due to the syntax overlap with page links.

I've just added a comment detailing one possible workaround on Linux: https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/513#issuecomment-1207193023 - the `ddg-open.sh` linked [here](https://unix.stackexchange.com/questions/497146/create-a-custom-url-protocol-handler/497147#497147) should be trivial to modify to fit your needs.

Alternatively, you may streamline your workflow by adding a key binding for your custom tool (and some regex to your script to remove the need of intricate highlighting, if you haven't already - you can also bash `grep` in there so that selecting any unique part of the line will do), or using a clipboard manager that triggers automatically when the selected text contains a phone number (& some keyword to prevent unintended calls). Either may be rigged up to splash a dialog (using eg. Zenity) if a given page or selection contains more than one valid number, or to confirm the call.

Happy hacking!

**Comment by Fooseigh** [comment, marked as answer]
Hi,

thank you for your advice. That will help to get my task solved.

I did not know about tel: URIs. It seems I already have one, leading to my softphone. Calling actually does not work, but the problem doesn' t seem to be related to ZIM.

Best regards,

Hans

**Comment by jaap-karssenberg** [comment]
URI s are not really supported, however it is really easy to define a
custom URL handler. So if you use "tel://" as a URL prefix and define your
script as the application to open those URLs it should work just fine.

(Also you can have a plugin intercept specific URLs and define an handler
within the application, but that does not seem to help here.)

--- Jaap
